pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -210,3 +219,85 @@ def visit_comprehension(self, node: astroid.Comprehension) -> None:
 
                 self.add_message("consider-using-dict-items", node=node)
                 return
+
+    @utils.check_messages("consider-using-str-partition")
+    def visit_subscript(self, node: astroid.Subscript) -> None:
+        """Add message when accessing first or last elements of a str.split()."""
+        assignment = None
+        subscripted_object = node.value
+        if isinstance(subscripted_object, astroid.Name):
+            # Check assignment of this subscripted object
+            assignment_lookup_results = node.value.lookup(node.value.name)
+            if (
+                len(assignment_lookup_results) == 0
+                or len(assignment_lookup_results[-1]) == 0
+            ):
+                return
+            assign_name = assignment_lookup_results[-1][-1]
+            if not isinstance(assign_name, astroid.AssignName):
+                return
+            assignment = assign_name.parent
+            if not isinstance(assignment, astroid.Assign):
+                return
+            # Set subscripted_object to the assignment RHS
+            subscripted_object = assignment.value
+
+        if isinstance(subscripted_object, astroid.Call):
+            # Check if call is .split() or .rsplit()
+            if isinstance(
+                subscripted_object.func, astroid.Attribute
+            ) and subscripted_object.func.attrname in ["split", "rsplit"]:
+                inferred = utils.safe_infer(subscripted_object.func)
+                if not isinstance(inferred, astroid.BoundMethod):
+                    return
+
+                # Check if subscript is 1 or -1
+                value = node.slice
+                if isinstance(value, astroid.Index):
+                    value = value.value
+
+                if isinstance(value, (astroid.UnaryOp, astroid.Const)):
+                    const = utils.safe_infer(value)
+                    const = cast(astroid.Const, const)
+                    p = re.compile(r"r*split")
+                    if const.value == -1:
+                        help_text = p.sub("rpartition", subscripted_object.as_string())
+                        self.add_message(
+                            "consider-using-str-partition",
+                            node=node,
+                            args=(help_text, -1),
+                        )
+                    elif const.value == 0:
+                        help_text = p.sub("partition", subscripted_object.as_string())
+                        self.add_message(
+                            "consider-using-str-partition",
+                            node=node,
+                            args=(help_text, 0),
+                        )
+                # Check if subscript is len(split) - 1
```

Comment:
Accessing the last element with `len(split) - 1` is an antipattern itself for which we could probably add another check as well (later) 😄 It will only add additional complexity which isn't worth it to maintain. Thus I would recommend to remove it.